alacritty_terminal/src/term/search.rs

```diff
@@ -0,0 +1,794 @@
+use std::cmp::min;
+use std::mem;
+use std::ops::RangeInclusive;
+
+use regex_automata::{dense, DenseDFA, Error as RegexError, DFA};
+
+use crate::grid::{BidirectionalIterator, Dimensions, GridIterator};
+use crate::index::{Boundary, Column, Direction, Point, Side};
+use crate::term::cell::{Cell, Flags};
+use crate::term::Term;
+
+/// Used to match equal brackets, when performing a bracket-pair selection.
+const BRACKET_PAIRS: [(char, char); 4] = [('(', ')'), ('[', ']'), ('{', '}'), ('<', '>')];
```

Comment:
I really wonder whether we can add `"` and `'` and maybe some other semantic escape chars here as well? Like I don't why not. Probably offtopic here.

Reply:
Nesting with quotes is a bit more complicated (read as impossible), but yeah this has nothing to do with this PR.